Chore: Upgrade Go to 1.19.1

sakjur opened the pull request:
**What this PR does / why we need it**:

Go 1.17 isn't supported anymore.

**Special notes for your reviewer**:

Progress:
- [x] Update Docker image templates
- [x] Upload new build container
- [ ] ~Update grabpl~ (I don't think we actually need to update grabpl itself)

sakjur committed:
WIP

sakjur committed:
Set public_suffix to a pre Ruby 2.6 version

sakjur committed:
we don't need to install python

sakjur committed:
Stretch->Buster

sakjur committed:
Bump versions in lib.star

sakjur committed:
Merge remote-tracking branch 'origin/main' into emil/202209/go1.19.1

sakjur committed:
Manually update linter

Sort of messy, but the .mod-file need to contain all dependencies that
use 1.16+ features, otherwise they're assumed to be compiled with
-lang=go1.16 and cannot access generics et al.

Bingo doesn't seem to understand that, but it's possible to manually
update things to get Bingo happy.

sakjur committed:
Merge remote-tracking branch 'origin/main' into emil/202209/go1.19.1

# Conflicts:
#	.drone.yml
#	scripts/drone/steps/lib.star

sakjur committed:
undo reformatting

sakjur committed:
Various lint improvements

sakjur committed:
More from the linter

sakjur committed:
goimports -w ./pkg/

sakjur committed:
Disable gocritic

sakjur committed:
Add/modify linter exceptions

sakjur commented:
I've removed the backporting goal, I think it's more realistic to make a separate PR for older releases with much less stringent linter requirements.

marefr commented on pkg/api/apikey.go:
```diff
@@ -95,7 +95,7 @@ func (hs *HTTPServer) DeleteAPIKey(c *models.ReqContext) response.Response {
 // Creates an API key.
 //
-// Will return details of the created API key
+// # Will return details of the created API key
```
Probably should re-generate swagger due to these changes to verify it still works as expected?

sakjur replied:
I was working with Sofia on getting this working a little bit better, but we agreed that even _if_ merging this would break Swagger it's worth getting these changes in sooner rather than later and fix it as it happens.

Unfortunately, the Go team decided to change the rules for comment formatting in a way that's rather disruptive and integrated in Go fmt, so not adhering to them leads to linter complaints. I'm not sure I think it was a bad decision seeing as Go comments have been rather lacking, but it's a nuisance.

marefr reviewed (commented):
Looks good I think. Added a comment to consider

sakjur committed:
Merge remote-tracking branch 'origin/main' into emil/202209/go1.19.1

# Conflicts:
#	pkg/api/alerting.go
#	pkg/api/annotations.go
#	pkg/api/api.go
#	pkg/api/apikey.go
#	pkg/api/metrics.go
#	pkg/api/org.go
#	pkg/api/org_users.go
#	pkg/api/quota.go
#	pkg/api/search.go
#	pkg/services/ngalert/state/manager.go

sakjur committed:
lint + flatten nested list

Go 1.19 doesn't support nested lists, and there wasn't an obvious workaround.
https://go.dev/doc/comment#lists

yuri-tceretian commented on pkg/services/ngalert/api/tooling/definitions/api.go:
```diff
@@ -3,23 +3,22 @@
-//      type: basic
+//	SecurityDefinitions:
+//	basic:
+//	  type: basic
```
The new formatting rules apparently broke the codegen in ngalert which treats this comment as yaml and does not like tabs
When I run `make` in the tooling dir it fails with `yaml: found character that cannot start any token`

sakjur replied:
That's an upstream issue that was fixed in a later version of the Swagger client, #55159 should make that work again.

It seems to be a bit confused as to the types though, so I haven't added updated Swagger definitions to that PR.

CoreMoni commented:
Can you please also update the go version for Grafana 8.X.X ?
